OrbitCore/UserDefinedCaptureData.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "UserDefinedCaptureData.h"
+
+#include "OrbitClientData/FunctionUtils.h"
+
+void UserDefinedCaptureData::InsertFrameTrack(const orbit_client_protos::FunctionInfo& function) {
```

Comment:
Maybe make it bool and document when it returns false, or ErrorMessageOr because nobody reads logs.

Same for Erase

Reply:
I'm not sure how to best handle this case. If I return success/error state here, what would the caller do? I can only think of printing a log message, which just moves the problem to a different place. I don't want this to bubble up to become a user visible error.

In both cases, it's not really a problem to call these functions multiple times with the same function, but it hints at a problem in the logic that calls these functions. 

I'm considering just removing the error check and allowing multiple calls to insert/erase with the same function (where second and subsequent calls will have no effect). 

WDYT?

Reply:
yeah - this is why I was thinking about returning bool here - caller may choose and in most cases will ignore it but there is still signal if needed. 

Returns true if the value was inserted and false otherwise (in the case when Function Set already contains this function)

Reply:
(in which case you do not need a check - you can return the second result of the std::set::insert)

Reply:
Can this actually happen, or is the context-menu only shown, if the function is not yet added?
If this is the case, I would actually argue that it should be a CHECK here, as we may have wrong assumptions.
If not, we might think of making it that way.

Reply:
I had a DCHECK here first and would prefer something like that. The logic around prevents this. However, it's strictly not a problem here, it's just a problem in the logic in the calling code. So maybe the right thing is to do a mix of what @dimitry- and you are suggesting. Return a boolean indicating whether the insertion / erasure was successful and CHECK by the caller (where the logic error happens).

Reply:
Ok, I now added CHECK directly here, as it felt artificial to return the result and then CHECK further out. Let me know if I should change this.

Reply:
Both would be fine for me. I'll leave it to @dimitry-.

Reply:
PTAL